datafusion/sql/src/unparser/expr.rs

```diff
@@ -819,8 +820,49 @@ impl Unparser<'_> {
             ScalarValue::TimestampMicrosecond(None, _) => {
                 Ok(ast::Expr::Value(ast::Value::Null))
             }
-            ScalarValue::TimestampNanosecond(Some(_ts), _) => {
-                not_impl_err!("Unsupported scalar: {v:?}")
+            ScalarValue::TimestampNanosecond(Some(_ts), tz) => {
+                if let Some(tz) = tz {
+                    let result = v
+                        .to_array()?
+                        .as_any()
+                        .downcast_ref::<TimestampNanosecondArray>()
+                        .ok_or(internal_datafusion_err!(
+                            "Unable to downcast to TimestampNanosecond from TimestampNanosecond scalar"
+                        ))?
+                        .value_as_datetime_with_tz(0, tz.parse()?)
+                        .ok_or(internal_datafusion_err!(
+                            "Unable to convert TimestampNanosecond to DateTime"
+                        ))?;
+                    Ok(ast::Expr::Cast {
+                        kind: ast::CastKind::Cast,
+                        expr: Box::new(ast::Expr::Value(ast::Value::SingleQuotedString(
+                            result.to_string(),
+                        ))),
+                        data_type: ast::DataType::Timestamp(None, TimezoneInfo::None),
+                        format: None,
+                    })
```

Comment:
I think this code is repeated in both branches -- we could probably avoid the replication

Reply:
Thanks, @alamb, for reminding me of this point. I forgot to refactor this part.